File: server/src/main/java/com/krillsson/sysapi/core/metrics/MotherboardMetrics.java
package com.krillsson.sysapi.core.metrics;

import com.krillsson.sysapi.core.domain.motherboard.Motherboard;
import com.krillsson.sysapi.core.domain.sensors.HealthData;

import java.util.List;

public interface MotherboardMetrics {
    Motherboard motherboard();

    List<HealthData> motherboardHealth();
}

File: src/main/kotlin/com/krillsson/sysapi/core/metrics/MotherboardMetrics.kt
package com.krillsson.sysapi.core.metrics

import com.krillsson.sysapi.core.domain.motherboard.Motherboard
import com.krillsson.sysapi.core.domain.sensors.HealthData

interface MotherboardMetrics {
    fun motherboard(): Motherboard
    fun motherboardHealth(): List<HealthData>
}